### PR TITLE
Switch button

### PR DESCRIPTION
**Pull request by vycdev**

### CODE REVIEW

File: packages/web/src/common/design/components/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React from "react"
+import { styled } from "../../../modules/theming/themes"
+import { Theme } from "../../../modules/theming/types/Theme"
+
+const size = 30
+
+interface checkboxProps {
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx]: interfaces should start with a capital char
```suggestion
interface CheckboxProps {
```

File: packages/web/src/common/design/components/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    borderRadius: "100px",
+
+    "&:before": {
+        position: "absolute",
+        content: `""`,
+        height: `${(size * 2 - 4) / 2}px`,
+        width: `${(size * 2 - 4) / 2}px`,
+        left: `2px`,
+        bottom: `1px`,
+        backgroundColor: "white",
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:57]: can u write this in terms of the current theme?

File: packages/web/src/common/design/components/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}))
+
+const SpanSwitch = styled.span(({ theme }) => ({
+    position: "absolute",
+    cursor: "pointer",
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0,
+    background: "#ccc",
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:45]: can u write this in terms of the current theme?

File: packages/web/src/common/design/components/Switch.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        transform: `translateX(${(size * 2 - 4) / 2}px)`
+    },
+    "span:before": {
+        borderRadius: "50%"
+    }
+}))
+
+const SpanSwitch = styled.span(({ theme }) => ({
+    position: "absolute",
+    cursor: "pointer",
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:40]: nice

File: packages/web/src/common/design/components/Switch.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        backgroundColor: "white",
+        WebkitTransition: theme.durations.short,
+        transition: theme.durations.short,
+        borderRadius: "50%"
+    }
+}))
+
+/**
+ * TODO:
+ * Somehow get the value idk if I want to do that
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:66]: just dont:)

File: packages/web/src/common/design/components/Switch.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        bottom: `1px`,
+        backgroundColor: "white",
+        WebkitTransition: theme.durations.short,
+        transition: theme.durations.short,
+        borderRadius: "50%"
+    }
+}))
+
+/**
+ * TODO:
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:65]: finally

File: packages/web/src/common/design/components/Switch.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+ * Somehow get the value idk if I want to do that
+ *
+ * Make a size prop
+ */
+
```
Comment (prescientmoon) [inline packages/web/src/common/design/components/Switch.tsx:70]: why use jsdoc in here?

**Review by prescientmoon — CHANGES_REQUESTED** [review]
Getting close

File: packages/web/src/common/design/components/Switch.tsx (hunk 7 above)
Comment (vycdev) [reply packages/web/src/common/design/components/Switch.tsx:70]: lol idk

**Pull request closed without merging**